mscore/scoretab.cpp

```diff
@@ -42,6 +42,31 @@
 
 namespace Ms {
 
+//---------------------------------------------------------
+//   MsTabBar::mousePressEvent
+//---------------------------------------------------------
+
+void MsTabBar::mousePressEvent(QMouseEvent* e)
+      {
+      QTabBar::mousePressEvent(e);
+      if (e->button() == Qt::MiddleButton)
+            _middleClickedTab = tabAt(e->pos());
+      }
+
+//---------------------------------------------------------
+//   MsTabBar::mouseReleaseEvent
+//---------------------------------------------------------
+
+void MsTabBar::mouseReleaseEvent(QMouseEvent* e)
+      {
+      QTabBar::mouseReleaseEvent(e);
+      if (e->button() == Qt::MiddleButton) {
+            if (tabAt(e->pos()) == _middleClickedTab)
```

Comment:
It would be nice to have a gesture recognizer that would detect actions and already respond to them. This is not a suggestion to do in this PR, just thoughts.  

```
MsTabBar::MsTabBar() {

_gestureRecognizer = new GestureRecognizer(this); <- setting event filter
connect(_gestureRecognizer, &GestureRecognizer::middleClicked, this, &MsTabBar::onMiddleClicked);
}
```